src/lib/scripts/automatic-prs/edit-pools-file.ts

```diff
@@ -0,0 +1,37 @@
+import * as fs from 'fs';
+
+export function allowListPool({
+  network,
+  poolType,
+  poolId,
+  poolDescription = '',
+}) {
+  const filePath = `./src/lib/config/${network}/pools.ts`;
+
+  const fileContent = fs.readFileSync(filePath, 'utf-8');
+
+  // Looking for the last element in the array [poolType] > Allowlist
+  const regex = new RegExp(
+    `${poolType}:[\\s\\n]*{[^}]*AllowList:[\\s\n]*\\[([\\s\\S]+?)\\]`
+  );
+
+  const match = fileContent.match(regex);
+
+  if (!match) throw Error('Pool type not found');
+
+  const arrayContent = match[1].trim();
+
+  if (arrayContent.includes(poolId))
+    throw Error(`${poolId} is already allowlisted`);
+
+  let updatedArrayContent = `${arrayContent} \n'${poolId}',`;
+
+  if (poolDescription) updatedArrayContent += ` // ${poolDescription} \n`;
+
+  // Replaces file with updated content:
+  const updatedContent = fileContent.replace(arrayContent, updatedArrayContent);
+
+  fs.writeFileSync(filePath, updatedContent);
+
+  console.log(`💾 ${network}/pools.ts file updated.`);
+}
```

Comment:
I don't think we need to have two files for this, why not just keep everything in allowlist-pool.ts?

Reply:
In my initial script I had other steps to allow running it from outside Github Actions, that's why I extracted the file edition part.
In any case, I think that having command line + edition responsibilities separated does not damage and improves readability so having the function inlined vs having the function in a separate file is a matter of taste.